apps/docs/content/guides/auth/third-party/clerk.mdx

```diff
@@ -0,0 +1,131 @@
+
+Follow Clerk's guide on [customizing the session token](https://clerk.com/docs/backend-requests/custom-session-token) to add or modify the `role` claim based on your project's needs.
+
+Once all Clerk session tokens for your instance contain the `role` claim, add a [new Third-Party Auth integration with Clerk](/dashboard/project/_/auth/third-party) in the Supabase dashboard or register it in the CLI as instructed above.
+
+## Setup the Supabase client library
+
+<Tabs type="underlined" queryGroup="clerk-create-client">
+
+<TabPanel id="ts" label="TypeScript">
+
+```typescript
+import { createClient } from '@supabase/supabase-js'
+
+const supabase = createClient('https://<supabase-project>.supabase.co', 'SUPABASE_ANON_KEY', {
+  accessToken: () => {
+    return Clerk.session?.getToken()
+  },
+})
+```
+
+</TabPanel>
+
+<TabPanel id="dart" label="Flutter">
+
+```dart
+TODO
+```
+
+</TabPanel>
+
+<TabPanel id="swift" label="Swift (iOS)">
+
+```swift
+import Clerk
+import Supabase
+
+let supabase = SupabaseClient(
+  supabaseURL: URL(string: "https://project-ref.supabase.io")!,
+  supabaseKey: "supabase.anon.key",
+  options: SupabaseClientOptions(
+    auth: SupabaseClientOptions.AuthOptions(
+      accessToken: {
+        try await Clerk.shared.session?.getToken()?.jwt
+      }
+    )
+  )
+)
+```
+
+</TabPanel>
+
+</Tabs>
+
+## Using RLS policies
+
+Once you've configured the Supabase client library to use Clerk session tokens, you can use RLS policies to secure access to your project's database, Storage objects and Realtime channels.
+
+The recommended way to design RLS policies with Clerk is to use claims present in your Clerk session token to allow or reject access to your project's data. Check [Clerk's docs](https://clerk.com/docs/backend-requests/resources/session-tokens) on the available JWT claims and their values.
+
+### Example: Check user organization role
+
+```sql
+create policy "Only organization admins can insert in table"
+  on table_name
+  to authenticated
+  with check (
+    (auth.jwt()->>'org_role' = 'org:admin')
+      and
+    (organization_id = auth.jwt()->>'org_id')
+  );
+```
+
+This RLS policy checks that the newly inserted row in the table has the user's declared organization ID (from the `org_id` session token claim) in the `organization_id` column. Additionally it ensures that they're an `org:admin`.
+
+This way only organization admins can add rows to the table, for organizations they're a member of.
+
+### Example: Check user has passed second factor verification
+
+```sql
+create policy "Only users that have passed second factor verification can read from table"
+  on table_name
+  as restrictive
+  to authenticated
+  using (
+    (auth.jwt()->'fva'->>1 != '-1')
+  );
+```
+
+This example uses a restrictive RLS policy checks that the second factor verification age element in the `fva` claim is not `'-1'` indicating the user has passed throuh second factor verification.
```

Comment:
This is accurate. For context, the structure of the claim is `[<first factor verification age in minutes>, <second factor verification age in minutes>]`.

For context, some higher-level docs: https://clerk.com/docs/guides/reverification

Reply:
Added a link to that page.